lib/player.js

```diff
@@ -5196,6 +5197,111 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
+            this.setMaxHardwareResolution(7680, 4320);
+          } else if (webapis.productinfo.isUdPanelSupported &&
+              webapis.productinfo.isUdPanelSupported()) {
+            this.setMaxHardwareResolution(3840, 2160);
+          } else {
+            this.setMaxHardwareResolution(1920, 1080);
+          }
+        }
+      } catch (e) {
+        shaka.log.alwaysWarn('Tizen: Error detecting screen size, default ' +
+            'screen size 1920x1080.');
+        this.setMaxHardwareResolution(1920, 1080);
+      }
+    } else if (shaka.util.Platform.isXboxOne()) {
+      const protectionCapabilities =
+          new Windows.Media.Protection.ProtectionCapabilities();
+      const protectionResult =
+          Windows.Media.Protection.ProtectionCapabilityResult;
+      // isTypeSupported may return "maybe", which means the operation is not
+      // completed. This means we need to retry
+      // https://learn.microsoft.com/en-us/uwp/api/windows.media.protection.protectioncapabilityresult?view=winrt-22621
+      let result = null;
+      try {
+        const type =
+            'video/mp4;codecs="hvc1,mp4a";features="decode-res-x=3840,' +
+            'decode-res-y=2160,decode-bitrate=20000,decode-fps=30,' +
+            'decode-bpc=10,display-res-x=3840,display-res-y=2160,' +
+            'display-bpc=8"';
+        const keySystem = 'com.microsoft.playready.recommendation';
+        do {
+          result = protectionCapabilities.isTypeSupported(type, keySystem);
+        } while (result === protectionResult.maybe);
+      } catch (e) {
+        result = protectionResult.notSupported;
+      }
+      if (result === protectionResult.probably) {
+        this.setMaxHardwareResolution(3840, 2160);
+      } else {
+        this.setMaxHardwareResolution(1920, 1080);
+      }
+    } else if (shaka.util.Platform.isWebOS()) {
+      try {
+        const deviceInfo =
+        /** @type {{screenWidth: number, screenHeight: number}} */(
+            JSON.parse(window.PalmSystem.deviceInfo));
+        // WebOS has always been able to do 1080p.  Assume a 1080p limit.
+        const width = Math.max(1920, deviceInfo.screenWidth);
+        const height = Math.max(1080, deviceInfo.screenHeight);
+        this.setMaxHardwareResolution(width, height);
+      } catch (e) {
+        shaka.log.alwaysWarn('WebOS: Error detecting screen size, default ' +
+            'screen size 1920x1080.');
+        this.setMaxHardwareResolution(1920, 1080);
+      }
+    } else if (shaka.util.Platform.isHisense()) {
+      // eslint-disable-next-line new-cap
+      if (window.Hisense_Get4KSupportState &&
+          // eslint-disable-next-line new-cap
+          window.Hisense_Get4KSupportState()) {
+        this.setMaxHardwareResolution(3840, 2160);
+      } else {
+        this.setMaxHardwareResolution(1920, 1080);
+      }
+    }
+    // TODO: Detect the maximum resolution for PS4
+    // TODO: Detect the maximum resolution for PS5
+  }
```

Comment:
Maybe it could be moved to some Device Utilities? Main player class is already thousands lines long.

Reply:
Maybe it makes sense in shaka.util.Platform since those are per-platform limitations. What do you think?

Reply:
I'm going to work on adding support to PS4 and PS5 and will change the location of this code.

Reply:
shaka.util.Platform makes perfect sense!